### PR TITLE
Fix pattern issue with EffPotion

### PR DESCRIPTION
**Pull request by Fusezion**
### Description

Fixes an issue with how `EffPotion` interacted with patterns for `potioneffects` and `potioneffecttypes`

---
**Target Minecraft Versions:**  N/A
**Requirements:**  N/A
**Related Issues:**  #5117

### CODE REVIEW

**Comment by TheLimeGlass** [comment]
Have you tested adding a potioneffecttype. The typical way people use this effect. Also do it with and without a variable.

**Comment by Fusezion** [comment]
> Have you tested adding a potioneffecttype. The typical way people use this effect. Sure you changed the order, but does this mean that the type doesn't work now in a variable (as the issue had)
> 
> Also do it with and without a variable.
as far as testing has gone yes they are still detecting correctly
```applescript
command /potion:
	trigger:
		set {_potion} to potion effect of slowness of tier 5 for 10 seconds
		apply {_potion} to player
		apply potion effect of speed of tier 5 for 10 seconds to player
		apply haste of tier 5 to player for 10 seconds
		apply ambient strength of tier 5 to player for 10 seconds
		apply jump boost of tier 5 without particles to player for 10 seconds
```
made it broadcast the matched pattern and all showed correctly
image

**Review by TheLimeGlass — APPROVED** [review]
This class will most likely get completely recoded in the future with the potion pull request revamp.

The simplest fix solves it.

**Comment by TheLimeGlass** [comment]
I would suggest adding the %potioneffect% part to the potions test of it's not already. I know there is a potions test but I don't know if it has %potioneffect% tests.

**Comment by Fusezion** [comment]
> I would suggest adding the %potioneffect% part to the potions test if it's not already. I know there is a potions test but I don't know if it has %potioneffect% tests.

From what I can tell there's already is one for it `ExprPotionEffects.sk` would it be worth it to add a test for `EffPotion` instead?

**Comment by Fusezion** [comment]
Went a head and created a test for `EffPotion` if there's any changes that should be done let me know. never done a test before

File: src/test/skript/tests/syntaxes/effects/EffPotion.sk [hunk 1]
```diff
@@ -0,0 +1,15 @@
+test "potion effect":
+	spawn a pig at spawn of world "world"
+	set {_pig} to last spawned pig
+	set {_potion} to potion effect of slowness of tier 5 for 10 seconds
+	apply {_potion} to player
+	assert {_pig} has potion effect slowness with "pig failed to apply slowness"
+	apply potion effect of speed of tier 5 for 10 seconds to player
+	assert {_pig} has potion effect speed with "pig failed to apply speed"
+	apply haste of tier 5 to player for 10 seconds
```
Comment (TheLimeGlass) [inline src/test/skript/tests/syntaxes/effects/EffPotion.sk:9]: There is no player in a test.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

File: src/test/skript/tests/syntaxes/effects/EffPotion.sk (hunk 1 above)
Comment (TheLimeGlass) [reply src/test/skript/tests/syntaxes/effects/EffPotion.sk:9]: Appears more than once

Comment (Fusezion) [reply src/test/skript/tests/syntaxes/effects/EffPotion.sk:9]: oops, guess I didn't realize that one I'll fix it up rn

**Review by TheLimeGlass — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**